Improve federation support

christophfriedrich opened the pull request:
Make federation info actually available to the utilised vue-components `Collection`, `Collections`, `Processes`, `FileFormats`.

m-mohr commented on src/store/index.js:
```diff
@@ -174,6 +180,12 @@ export default new Vuex.Store({
 			let capabilities = cx.state.connection.capabilities();
 
+			// Note down federation things from capabilities
+			let federation = capabilities.listFederation();
```
fyi: You can obtain the object as such: `capabilities.toJSON().federation`

christophfriedrich replied:
Yeah but if there exists a function precisely for my use case then I kinda like to use it :D

m-mohr commented on src/store/index.js:
```diff
@@ -174,6 +180,12 @@ export default new Vuex.Store({
 
+			// Note down federation things from capabilities
+			let federation = capabilities.listFederation();
+			if (Object.keys(federation).length > 0) {   // empty object = no federation -> don't commit (leaving default value)
```
```suggestion
			if (federation.length > 0) {   // empty array = no federation -> don't commit (leaving default value)
```
listFederation always returns an array.

m-mohr commented:
Setting to draft until we have a new version number for the JS client so that we can also update the dependency in the package.json.

m-mohr commented:
Thanks. We can keep it as is for now, the ⚠️ is more a nice-to-have, I'd say.

With regards to the Federation and Missing notice:
- The federation notice is a bit weird as it only lists one backend. I think the federation notice should only be shown with > 1 backends in the list.
- We should probably change the color of the federation notice to something more neutral, the warning color is not really suitable. Maybe some blue/violet color that we already use (e.g. see the experimental notice).
- Having two boxes may be a little over the top, but is fine for now. The online/offline notice in the federation notice is not really needed though, als it's always offline of course. I'd remove it from there.
- We should probably improve the texts slightly:
  - Federation missing notice: `The following list is incomplete as the following services in the federation are currently not available: ...`
  - Federation notice: `This {{ entity }} is provided by the following services in the federation:</p>`

christophfriedrich commented:
Screenshots:

Example in modal:

![image](https://github.com/user-attachments/assets/5da6213f-0789-4df6-b2d6-e969489d3302)

Example in wizard:

![image](https://github.com/user-attachments/assets/a92fdfbd-51b7-4b47-811a-847f5e8b04d1)

Example in user content panel:

![image](https://github.com/user-attachments/assets/002adeee-05d5-40c5-9063-810725702930)

m-mohr commented on src/components/WorkPanelMixin.js:
```diff
@@ -23,6 +25,8 @@ export default (namespace, singular, plural, loadInitially = true) => {
 		},
 		computed: {
+			...Utils.mapState(['federation']),
+			...Utils.mapState(namespace, ['missing']),
 			...Utils.mapState(namespace, {data: namespace}),
 			...Utils.mapState(namespace, ['pages', 'hasMore']),
```
```suggestion
			...Utils.mapState(namespace, {data: namespace}),
			...Utils.mapState(namespace, ['missing', 'pages', 'hasMore']),
```

christophfriedrich replied:
I kinda liked keeping all federation stuff together (line 28 and 29) and separating it from the rest (first two lines federation, then the rest)

m-mohr replied:
I'd like to keep it as concise as possible to avoid potential confusion, I also had to look for a second to realize it's the same namespace. We also kept it together in all other instances.

christophfriedrich replied:
Addressed in 84f3e767873b56696b2c6bdbc029f389632b6dd5

m-mohr commented on src/store/storeFactory.js:
```diff
@@ -112,6 +112,7 @@ export default ({namespace, listFn, paginateFn, createFn, updateFn, deleteFn, re
 					for (let d of data) {
 						state[namespace].push(d);
 					}
+					state.missing = data['federation:missing'];  // yes, accessing a property on an array
```
The missing is not initiated in the getDefaultState method above, I could imagine that this could become an issue and lead to the issues you saw.

christophfriedrich replied:
Good spot!! I added it to getDefaultState and the weird behaviour indeed went away. Let's see if this fix is more permanent now 😆

christophfriedrich replied:
Fixed in 84f3e767873b56696b2c6bdbc029f389632b6dd5

christophfriedrich commented:
While taking the screenshots I realised that sometimes the FederationMissingNotice is displayed in the `orangered` colour it's supposed to have, sometimes in the `darkgoldenrod` of the FederationNotice:

![image](https://github.com/user-attachments/assets/a92fdfbd-51b7-4b47-811a-847f5e8b04d1)

![image](https://github.com/user-attachments/assets/49dd0e39-9885-4b16-ba18-f4839ae8fe8c)

I was able to reproduce: It is orange as long as no FederationNotice has been added to the DOM, then it changes to gold.

Probably need to adapt CSS specificity in FederationMissingNotice to overwrite more permanently, or not use the same class names for the two components in the first place. Could also make the style scoped? `<style lang="scss" scoped>...`

christophfriedrich commented:
The CSS issue I mentioned was tracked and fixed upstream in vue-components, and I just addressed the two comments from the review in a new commit -> ready to merge?

m-mohr commented:
Yes, looks, good thanks. I'll make release of the upstream components soon and then merge the open PRs along the way, thanks.

m-mohr commented:
@christophfriedrich
- openeo-js-client is available as v2.8.0
- openeo-vue-components will be available as v2.19.0 in a couple of minutes

christophfriedrich commented:
Will do and push once I confirmed locally that it works

christophfriedrich commented:
Confirmed, package.json updated
